src/components/game-board/GameBoard.tsx

```diff
@@ -55,52 +47,53 @@ function GameBoard({ className }: ChildProps): React.JSX.Element{
         }));
 
         navigate('/game-over');
-    }, [navigate, setGame]);
+    }, [navigate]);
 
-    const updatePlayerTurn =  useCallback(()=>{
-        setShowGameDialog(false);
-        setGame((prevGame: Game) => {
-            const currentPlayer = prevGame.activePlayer;
-            const currentPlayerIndex = prevGame.players.indexOf(currentPlayer!);
-            const nextPlayerIndex = (currentPlayerIndex + 1) % prevGame.players.length;
-            return {
-                ...prevGame,
-                totalTurns: prevGame.totalTurns - 1,
-                activePlayer: prevGame.players[nextPlayerIndex],
-                nextPlayer: prevGame.players[(nextPlayerIndex + 1) % prevGame.players.length],
+    const updatePlayerInsideGameObject = useCallback((prevGame: Game) => {
+        const currentPlayer = prevGame.activePlayer;
+        const currentPlayerIndex = prevGame.players.indexOf(currentPlayer!);
+        const nextPlayerIndex = (currentPlayerIndex + 1) % prevGame.players.length;
+        return {
+            ...prevGame,
+            totalTurns: prevGame.totalTurns - 1,
+            activePlayer: prevGame.players[nextPlayerIndex],
+            nextPlayer: prevGame.players[(nextPlayerIndex + 1) % prevGame.players.length],
 
-            };
-        });
-    }, []);
+        };
+    },[]);
+    const updatePlayerTurn = useCallback(() => {
+        setShowGameDialog(false);
```

Comment:
Is this really relevant to "updatePlayerTurn"? Isn't this more relevant to a "startGame" function?

Reply:
you are right , its redundant here